read customized values if they exists to make hook works.

adglkh opened the pull request:
Currently, it doesn't read settings from $SNAP_COMMON/wekan_settings.sh
which results in regarding customized settings doesn't take effect.
To verify it
sudo snap set wekan-ondra port=9876
sudo snap disable wekan-ondra
sudo snap enable wekan-ondra

Then navigate to localhost:9876 in your browser to check if you can open wekan website from you local.

adglkh committed:
read customized values if they exists to make hook works.

Currently, it doesn't read settings from $SNAP_COMMON/wekan_settings.sh
which results in regarding customized settings doesn't take effect.
e.g. port

adglkh committed:
Bump the version up.

kubiko reviewed (commented):
source $SNAP_COMMON/wekan_settings.sh should not be needed, as this is called from $SNAP/bin/config, check line 8. But export there is indeed missing.

kubiko reviewed (commented):
Actually export there is not needed either, as export is already done inside $SNAP_COMMON/wekan_settings.sh
But I just noticed that we export there also keys which were not set, and that is bug

adglkh committed:
Fixed description for configure hooks setting. The name of key should
be lower case instead of upper case. And revert the customized values
reading from wekan_settings.sh file.

adglkh commented:
If we export keys which were not set, they will be overridden with default key value. See $SNAP/wekan-read-settings file So that's not a bug.
But I suppose that end-user wrongly sets the configure hooks after reading the description by using the upper case key, there will be no chance for them to get it work.
I fixed them in description section in yaml file. Please have a check

adglkh commented:
Basically, if the keys which were not set will be overridden by using the default value.
So it's not a bug.
It turns out that if normal use sets configure hooks with upper case key which is actually should be lower case key. That's why it doesn't take effect.
Please take a look description in yaml file.

kubiko commented:
snapcraft yaml is out of sync. I will remove settings description from there completely, as now you can get all info by calling wekan.help. That is cleaner implementation as that reads from src/config key mapping rules. Like this we have central place where all key mapping and all other places are reading it from there.
Thanks for spotting this!
I also fixed now problem with handling empty keys, just did not have time to fix that before MWC :(

adglkh commented:
I like the idea of calling wekan.help to list all available key names from the command line. Indeed, It's cleaner than the hooks-fix branch implementation that just lists all of them in yaml file.
Could you please bump up the version and release the latest version into store?
Thanks.

kubiko commented:
I have it set on auto build, so all changes are already in the store. But I just noticed they have mad 0.12 release. So now building it
